native/effektio/src/api/invitation.rs

```diff
@@ -144,34 +142,29 @@ impl InvitationController {
         }
     }
 
-    pub async fn add_event_handler(&mut self, client: &MatrixClient) {
-        let invitations = self.get_invitations(client).await;
-        self.invitations.lock_mut().clone_from(&invitations);
+    pub fn add_event_handler(&mut self, client: &MatrixClient) {
         let me = self.clone();
 
-        client.add_event_handler_context(client.clone());
         client.add_event_handler_context(me.clone());
         let handle = client.add_event_handler(
             |ev: StrippedRoomMemberEvent,
              room: MatrixRoom,
-             Ctx(client): Ctx<MatrixClient>,
+             c: MatrixClient,
              Ctx(me): Ctx<InvitationController>| async move {
                 // user got invitation
-                me.clone().process_stripped_event(ev, room, &client).await;
+                me.clone().process_stripped_event(ev, room, &c).await;
             },
         );
         self.stripped_event_handle = Some(handle);
 
-        client.add_event_handler_context(client.clone());
         client.add_event_handler_context(me);
         let handle = client.add_event_handler(
-            |ev: OriginalSyncRoomMemberEvent,
+            |ev: SyncRoomMemberEvent,
              room: MatrixRoom,
-             Ctx(client): Ctx<MatrixClient>,
-             Ctx(me): Ctx<InvitationController>,
-             handle: EventHandlerHandle| async move {
+             c: MatrixClient,
+             Ctx(me): Ctx<InvitationController>| async move {
```

Comment:
if `Ctx(client): Ctx<MatrixClient>` can be written as `c: MatrixClient,` now, can't `Ctx(me): Ctx<InvitationController>` also be written as `me: InvitationController` now? (same for the change above)

Reply:
`client` is provided as default, but controller is provided by custom parameter.
Please read this code snippet: https://github.com/matrix-org/matrix-rust-sdk/blob/f8b02d1a11d07d28ce87132b93d30d6b03d76b5a/crates/matrix-sdk/src/event_handler/mod.rs#L422